rebac-admin-backend/v1/groups.go

```diff
@@ -1,11 +1,241 @@
+		response := h.GroupsErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	writeResponse(w, http.StatusOK, identity)
+}
+
+// PutGroupsItem updates the group identified by the provided ID.
+// (PUT /groups/{id})
+func (h handler) PutGroupsItem(w http.ResponseWriter, req *http.Request, id string) {
+	ctx := req.Context()
+
+	group := new(resources.Group)
+	defer req.Body.Close()
+
+	if err := json.NewDecoder(req.Body).Decode(group); err != nil {
+		writeErrorResponse(w, NewValidationError("request doesn't match the expected schema"))
+		return
+	}
+
+	if id != *group.Id {
+		writeErrorResponse(w, NewValidationError("group ID from path does not match the Group object"))
+		return
+	}
+
+	group, err := h.Groups.UpdateGroup(ctx, group)
+	if err != nil {
+		response := h.GroupsErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	writeResponse(w, http.StatusOK, group)
+}
+
+// GetGroupsItemEntitlements returns the list of entitlements for a group identified by the provided ID.
+// (GET /groups/{id}/entitlements)
+func (h handler) GetGroupsItemEntitlements(w http.ResponseWriter, req *http.Request, id string, params resources.GetGroupsItemEntitlementsParams) {
+	ctx := req.Context()
+
+	entitlements, err := h.Groups.GetGroupEntitlements(ctx, id, &params)
+	if err != nil {
+		response := h.GroupsErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	response := resources.GetGroupEntitlementsResponse{
+		Data:   entitlements,
+		Status: http.StatusOK,
+	}
+
+	writeResponse(w, http.StatusOK, response)
+}
+
+// PatchGroupsItemEntitlements Adds or removes entitlements to/from a group.
+// (PATCH /groups/{id}/entitlements)
+func (h handler) PatchGroupsItemEntitlements(w http.ResponseWriter, req *http.Request, id string) {
+	ctx := req.Context()
+
+	patchesRequest := new(resources.GroupEntitlementsPatchRequestBody)
+	defer req.Body.Close()
+
+	if err := json.NewDecoder(req.Body).Decode(patchesRequest); err != nil {
+		writeErrorResponse(w, NewValidationError("request doesn't match the expected schema"))
+		return
+	}
+
+	_, err := h.Groups.PatchGroupEntitlements(ctx, id, patchesRequest.Patches)
+	if err != nil {
+		response := h.GroupsErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+}
+
+// GetGroupsItemIdentities returns the list of identities within a group identified by given id.
+// (GET /groups/{id}/identities)
+func (h handler) GetGroupsItemIdentities(w http.ResponseWriter, req *http.Request, id string, params resources.GetGroupsItemIdentitiesParams) {
+	ctx := req.Context()
+
+	groups, err := h.Groups.GetGroupIdentities(ctx, id, &params)
+	if err != nil {
+		response := h.GroupsErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	response := resources.GetGroupIdentitiesResponse{
+		Data:   groups.Data,
+		Status: http.StatusOK,
```

Comment:
this seems a bit unusual

Reply:
Do you mean the status code being in the response type?

Reply:
I think it's about the `.Data`.
Right now we are keeping this "wrapper" because we will need the backend service to pass (along with the data results) the paging info, since the controller layer doesn't have a way to know, for example, which will be the next page, the total items available, or the next page token in case of token-based pagination.
Right now we only use the `Data` field, but with pagination we'll have to rely on backend service layers to provide those other pieces of info.